Python: add YAML support

aibaars opened the pull request.

RasmusWL reviewed (changes requested):
Can we rename `YAML.qll` to `Yaml.qll` please? (as you mentioned yourself offline) I think this better matches the naming scheme we want for the future, even if we diverge a little from what JS is doing right now :disappointed: 

I played around with my own database with YAML files from building the internal extractor ([here](https://github.com/RasmusWL/codeql/commit/a8d8786c452b573cfb39c545c42c12bf604169c9)). Turns out that currently YAML files are not extracted as part of running `codeql test`. I don't have a desire to block this PR before we have this ability, but something we should look at getting ASAP.

> I updated the dbscheme stats file by copying the YAML stats from JavaScript. If there is a better way of producing stats, please let me know.

Sounds good to me :+1: We don't have a good way to produce stats right now (we're a bit afraid that updating existing stats might break points-to :disappointed:)

---

References I looked at
- [shared implementation](https://github.com/github/codeql/blob/cff7f631933ab55fcee8224b5faad04200be4ec5/shared/yaml/codeql/yaml/Yaml.qll)
- [JS YAML tests](https://github.com/github/codeql/tree/cff7f631933ab55fcee8224b5faad04200be4ec5/javascript/ql/test/library-tests/YAML)

aibaars commented:
@RasmusWL I renamed the Yaml.qll library, implemented printAst support, and copied the Yaml tests from javascript with minor changes: 

```patch
diff -r javascript/ql/test/library-tests/Yaml/printAst.ql python/ql/test/library-tests/Yaml/printAst.ql
1,2c1,2
< import javascript
< import semmle.javascript.PrintAst
---
> import semmle.python.PrintAst
diff -r javascript/ql/test/library-tests/Yaml/tests.ql python/ql/test/library-tests/Yaml/tests.ql
1c1
< import javascript
---
> import semmle.python.Yaml
```

RasmusWL reviewed (approved):
Awesome :muscle:
